includes/Classifai/Providers/Azure/Language.php

```diff
@@ -0,0 +1,391 @@
+				'label_for'     => 'endpoint_url',
+				'input_type'    => 'text',
+				'default_value' => $settings['endpoint_url'],
+				'class'         => 'classifai-provider-field hidden provider-scope-' . static::ID, // Important to add this.
+				'description'   => sprintf(
+					wp_kses(
+						// translators: 1 - link to create a Language resource.
+						__( 'Azure Cognitive Service Language Endpoint, <a href="%1$s" target="_blank">create a Language resource</a> in the Azure portal to get your key and endpoint.', 'classifai' ),
+						array(
+							'a' => array(
+								'href'   => array(),
+								'target' => array(),
+							),
+						)
+					),
+					esc_url( 'https://portal.azure.com/#home' )
+				),
+			]
+		);
+	}
+
+	/**
+	 * Returns the default settings for this provider.
+	 *
+	 * @return array
+	 */
+	public function get_default_provider_settings(): array {
+		$common_settings = [
+			'api_key'       => '',
+			'endpoint_url'  => '',
+			'authenticated' => false,
+		];
+
+		return $common_settings;
+	}
+
+	/**
+	 * Sanitize the settings for this provider.
+	 *
+	 * Can also be useful to verify the Provider API connection
+	 * works as expected here, returning an error if needed.
+	 *
+	 * @param array $new_settings The settings array.
+	 * @return array
+	 */
+	public function sanitize_settings( array $new_settings ): array {
+		$settings = $this->feature_instance->get_settings();
+
+		$new_settings[ static::ID ]['api_key']       = sanitize_text_field( $new_settings[ static::ID ]['api_key'] ?? $settings[ static::ID ]['api_key'] );
+		$new_settings[ static::ID ]['endpoint_url']  = esc_url_raw( $new_settings[ static::ID ]['endpoint_url'] ?? $settings[ static::ID ]['endpoint_url'] );
+		$new_settings[ static::ID ]['authenticated'] = false;
+
+		if ( ! empty( $new_settings[ static::ID ]['endpoint_url'] ) && ! empty( $new_settings[ static::ID ]['api_key'] ) ) {
+			$new_settings[ static::ID ]['authenticated'] = $this->authenticate_credentials(
+				$new_settings[ static::ID ]['endpoint_url'],
+				$new_settings[ static::ID ]['api_key']
+			);
+		}
+
+		if ( ! $new_settings[ static::ID ]['authenticated'] ) {
+			add_settings_error(
+				'authenticated',
+				400,
+				esc_html( 'There was an error authenticating with Azure Language Services. Please check your credentials.' ),
+				'error'
+			);
+
+			// disable the feature if we're unable to authenticate./
+			$new_settings['status'] = 0;
+		}
+
+		return $new_settings;
+	}
+
+	/**
+	 * Authenticates our credentials.
+	 *
+	 * Performs a simple test to ensure the credentials are valid.
+	 *
+	 * @param string $url Endpoint URL.
+	 * @param string $api_key Api Key.
+	 *
+	 * @return bool|WP_Error
+	 */
+	protected function authenticate_credentials( string $url, string $api_key ) {
+		$rtn = false;
+
+		$endpoint = trailingslashit( $url ) . '/text/analytics/v3.1/languages';
+		$endpoint = add_query_arg( 'api-version', static::API_VERSION, $endpoint );
+
+		$request = safe_wp_remote_get(
```

Comment:
This is throwing error. Should it use `wp_remote_post()` instead?

Reply:
See this [comment](https://github.com/10up/classifai/pull/786#issuecomment-2269844794) and the follow-up comments for context. What error are you seeing?

Reply:
I am seeing the following when trying to connect to the service:

<img width="814" alt="Screenshot 2024-09-06 at 2 10 22 PM" src="https://github.com/user-attachments/assets/0cf92416-6b4c-4057-ac39-572c15a01676">

Reply:
Here's the neater version:

```
[06-Sep-2024 08:40:15 UTC] PHP Fatal error:  Uncaught TypeError: http_build_query(): Argument #1 ($data) must be of type array, string given in /Users/work/Sites/classifai/app/public/wp-includes/Requests/src/Transport/Curl.php:578
Stack trace:
#0 /Users/work/Sites/classifai/app/public/wp-includes/Requests/src/Transport/Curl.php(578): http_build_query('{"documents": [...', '', '&')
#1 /Users/work/Sites/classifai/app/public/wp-includes/Requests/src/Transport/Curl.php(391): WpOrg\Requests\Transport\Curl::format_get('https://classif...', '{"documents": [...')
#2 /Users/work/Sites/classifai/app/public/wp-includes/Requests/src/Transport/Curl.php(171): WpOrg\Requests\Transport\Curl->setup_handle('https://classif...', Array, '{"documents": [...', Array)
#3 /Users/work/Sites/classifai/app/public/wp-includes/Requests/src/Requests.php(469): WpOrg\Requests\Transport\Curl->request('https://classif...', Array, '{"documents": [...', Array)
#4 /Users/work/Sites/classifai/app/public/wp-includes/class-wp-http.php(397): WpOrg\Requests\Requests::request('https://classif...', Array, '{"documents": [...', 'GET', Array)
#5 /Users/work/Sites/classifai/app/public/wp-includes/class-wp-http.php(638): WP_Http->request('https://classif...', Array)
#6 /Users/work/Sites/classifai/app/public/wp-includes/http.php(184): WP_Http->get('https://classif...', Array)
#7 /Users/work/Sites/classifai/app/public/wp-content/plugins/classifai/includes/Classifai/Helpers.php(686): wp_remote_get('https://classif...', Array)
#8 /Users/work/Sites/classifai/app/public/wp-content/plugins/classifai/includes/Classifai/Providers/Azure/Language.php(164): Classifai\safe_wp_remote_get('https://classif...', Array)
#9 /Users/work/Sites/classifai/app/public/wp-content/plugins/classifai/includes/Classifai/Providers/Azure/Language.php(122): Classifai\Providers\Azure\Language->authenticate_credentials('https://classif...', '95193e52eec647f...')
#10 /Users/work/Sites/classifai/app/public/wp-content/plugins/classifai/includes/Classifai/Features/Feature.php(258): Classifai\Providers\Azure\Language->sanitize_settings(Array)
#11 /Users/work/Sites/classifai/app/public/wp-includes/class-wp-hook.php(326): Classifai\Features\Feature->sanitize_settings(Array)
#12 /Users/work/Sites/classifai/app/public/wp-includes/plugin.php(205): WP_Hook->apply_filters(Array, Array)
#13 /Users/work/Sites/classifai/app/public/wp-includes/formatting.php(5115): apply_filters('sanitize_option...', Array, 'classifai_featu...', Array)
#14 /Users/work/Sites/classifai/app/public/wp-includes/option.php(866): sanitize_option('classifai_featu...', Array)
#15 /Users/work/Sites/classifai/app/public/wp-admin/options.php(344): update_option('classifai_featu...', Array)
#16 {main}
  thrown in /Users/work/Sites/classifai/app/public/wp-includes/Requests/src/Transport/Curl.php on line 578

```